modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageStore.java

```diff
@@ -222,12 +298,22 @@ public void addMessageToExpiryQueue(Long messageId, Long expirationTime,
                                         boolean isMessageForTopic, String destination)
             throws AndesException;
 
-    /***
-     * Store level method to remove all metadata references to all messages addressed to a specific queue.
+    /**
+     * Store level method to remove all metadata addressed to a specific queue.
+     *
      * @param storageQueueName name of the queue being purged
+     * @return the number of messages that were deleted
      * @throws AndesException
      */
-    public void deleteAllMessageMetadata(String storageQueueName) throws AndesException;
+    public int deleteAllMessageMetadata(String storageQueueName) throws AndesException;
+
+    /**
+     * Store level method to remove all metadata in a dead letter channel
+     *
+     * @param dlcQueueName name of the queue being purged
+     * @throws AndesException
+     */
+    public int clearDlcQueue(String dlcQueueName) throws AndesException;
```

Comment:
In some method signatures "Dlc" is used as "DLC", shall we use only one of those?